src/hclib-runtime.c

```diff
@@ -56,6 +55,10 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <hclib-module.h>
 #include <hclib-instrument.h>
 
+#ifdef HAVE_DLFCN_H
```

Comment:
@srirajpaul where is HAVE_DLFCN_H set?

Reply:
inside  hclib-install/include/hclib_config.h when HClib is build.

Reply:
@srirajpaul is this a built-in autoconf thing? I'm only asking because I see you've added special commands for AIO but not for this. I realize this is probably a dumb question due to my autoconf ignorance, just want to make sure we aren't missing anything

Reply:
@agrippa  DLFCN was already supported, as in the installed hclib-install/include/hclib_config.h was already setting HAVE_DLFCN_H to 1 if elfin.h is present.

This was happening because https://github.com/habanero-rice/hclib/blob/32ada897f673edd56375e83a58c1a8cfd5ba5dcd/configure.ac#L21 was using a macro AC_PROG_LIBTOOL to enable support for shared libraries. This was automatically adding the AC_CHECK_HEADERS([dlfcn.h]..) and together with HAVE_DLFCN_H  in https://github.com/habanero-rice/hclib/blob/32ada897f673edd56375e83a58c1a8cfd5ba5dcd/inc/hclib_config.h.in#L7 sets the HAVE_DLFCN_H macro to 1 in hclib_config.h  if dlfcn.h is present.
The question is not dump. I am also learning more about autoconf through this and it took some time to find out how DLFCN was getting set.